src/csvcubed/inspect/browsertable.py

```diff
@@ -0,0 +1,175 @@
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import List, OrderedDict, Union
+
+import uritemplate
+
+from csvcubed.inspect.browsercolumns import (
+    AttributeColumn,
+    DataCubeColumn,
+    DimensionColumn,
+    MeasuresColumn,
+    PivotedObservationsColumn,
+    StandardShapeObservationsColumn,
+    SuppressedColumn,
+    UnitsColumn,
+)
+from csvcubed.inspect.browsers import MetadataBrowser, TableBrowser
+from csvcubed.inspect.lazyfuncdescriptor import lazy_func_field
+from csvcubed.models.cube.cube_shape import CubeShape
+from csvcubed.models.sparqlresults import CatalogMetadataResult
+from csvcubed.utils.iterables import single
+from csvcubed.utils.qb.components import EndUserColumnType
+from csvcubed.utils.sparql_handler.code_list_inspector import CodeListInspector
+from csvcubed.utils.sparql_handler.column_component_info import ColumnComponentInfo
+from csvcubed.utils.sparql_handler.csvw_inspector import CsvWInspector
+from csvcubed.utils.sparql_handler.data_cube_inspector import DataCubeInspector
+from csvcubed.utils.tableschema import CsvWRdfManager
+
+
+@dataclass(frozen=True)
+class DataCubeTable(MetadataBrowser, TableBrowser):
+    def _get_shape(self) -> CubeShape:
+        return self.data_cube_inspector.get_shape_for_csv(self.csv_url)
+
+    def _get_dataset_uri(self) -> str:
+        return self.data_cube_inspector.get_cube_identifiers_for_csv(
+            self.csv_url
+        ).data_set_url
+
+    def _get_columns(self) -> OrderedDict[str, DataCubeColumn]:
+        columns = OrderedDict[str, DataCubeColumn]()
+        for c in self.data_cube_inspector.get_column_component_info(self.csv_url):
+            if c.column_definition.title is not None:
+                columns[c.column_definition.title] = _map_to_data_cube_column(self, c)
+
+        return columns
+
+    shape: CubeShape = lazy_func_field(_get_shape)
+    data_set_uri: str = lazy_func_field(_get_dataset_uri)
+    columns: OrderedDict[str, DataCubeColumn] = lazy_func_field(
+        _get_columns, repr=False
+    )
+
+    def __hash__(self):
```

Comment:
You two comfortable with what the hash function is for?

Reply:
As in, generally, or in this specific case?

Reply:
Ideally both.

Reply:
Something to do with checking for equality?